dist/docker/Dockerfile

```diff
@@ -1,37 +1,47 @@
-
+# image for building
 FROM alpine:latest AS builder
 
-ARG BUILD_TYPE
-ARG RELEASE
-
-RUN if [ $RELEASE = "master" ] ; \
-    then \
-      wget https://github.com/qbittorrent/qBittorrent/archive/refs/heads/master.zip && \
-      unzip master.zip && \
-      cd qBittorrent-master ; \
-    else \
-      wget https://github.com/qbittorrent/qBittorrent/archive/refs/tags/release-${RELEASE}.tar.gz && \
-      tar xf release-${RELEASE}.tar.gz && \
-      cd qBittorrent-release-${RELEASE} ; \
-    fi && \
-    apk add --no-cache qt6-qttools-dev g++ libtorrent-rasterbar-dev cmake boost-dev ninja && \
-    cmake -B build-nox -G "Ninja" -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DGUI=OFF -DQT6=ON -DSTACKTRACE=OFF && \
-    cmake --build build-nox && \
-    cmake --build build-nox --target install/strip
-
+ARG QBT_VERSION
+
+# alpine linux qbittorrent package: https://git.alpinelinux.org/aports/tree/community/qbittorrent/APKBUILD
+
+RUN \
+  apk --update-cache add \
+    boost-dev \
+    cmake \
+    g++ \
+    libtorrent-rasterbar-dev \
+    ninja \
+    qt6-qtbase-dev \
+    qt6-qttools-dev && \
+  if [ "$QBT_VERSION" = "git" ]; then \
+    wget https://github.com/qbittorrent/qBittorrent/archive/refs/heads/master.zip && \
+    unzip master.zip && \
+    cd qBittorrent-master ; \
+  else \
+    wget "https://github.com/qbittorrent/qBittorrent/archive/refs/tags/release-${QBT_VERSION}.tar.gz" && \
+    tar -xf "release-${QBT_VERSION}.tar.gz" && \
+    cd "qBittorrent-release-${QBT_VERSION}" ; \
+  fi && \
+  cmake \
+    -B build \
+    -G Ninja \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    -DGUI=OFF \
+    -DQT6=ON \
+    -DSTACKTRACE=OFF && \
+  cmake --build build && \
+  cmake --install build
```

Comment:
For improved caching, it would be worth moving the `apk add` into its own `RUN` step.